src/Microsoft.AspNetCore.Http.Abstractions/Internal/ParsingHelpers.cs

```diff
@@ -26,7 +26,7 @@ private static IEnumerable<string> GetHeaderSplitImplementation(StringValues val
         {
             foreach (var segment in new HeaderSegmentCollection(values))
             {
-                if (segment.Data.HasValue)
+                if (segment.Data.HasValue && !string.IsNullOrWhiteSpace(segment.Data.Value))
```

Comment:
We have a separate bug to replace all IsNullOrWhiteSpace calls with IsNullOrEmpty. IsNullOrWhiteSpace checks for a bunch of unicode characters we don't want to check.